src/precice/SolverInterface.cpp

```diff
@@ -35,7 +37,7 @@ SolverInterface::SolverInterface(
     int                    solverProcessIndex,
     int                    solverProcessSize,
     void *                 communicator)
-    : _impl(new impl::SolverInterfaceImpl(toSV(participantName), toSV(configurationFileName), solverProcessIndex, solverProcessSize, communicator))
+    : _impl(new impl::SolverInterfaceImpl(toSV(participantName), toSV(configurationFileName), solverProcessIndex, solverProcessSize, {communicator}))
```

Comment:
I have not used `std::optional` before, but what is the meaning of the curly braces here?

Reply:
They mean "construct what is required here using `communicator`". It's a shortcut for `std::optional<void*>{communicator}`.

Also very useful in return statements of functions.